website/server/libs/payments.js

```diff
@@ -125,13 +130,39 @@ api.addSubToGroupUser = async function addSubToGroupUser (member, group) {
         {name: 'EMAIL', content: getUserInfo(member, ['email']).email},
         {name: 'PAYMENT_METHOD', content: memberPlan.paymentMethod},
         {name: 'PURCHASED_PLAN', content: JSON.stringify(memberPlan)},
-        {name: 'ACTION_NEEDED', content: 'User has joined group plan. Tell them to cancel subscription then give them free sub.'},
+        {name: 'ACTION_NEEDED', content: 'User has joined group plan and has been told to cancel their subscription then email us. Ensure they do that then give them free sub.'},
+        // TODO User won't get email instructions if they've opted out of all emails. See if we can make this email an exception and if not, report here whether they've opted out.
       ]);
     }
 
-    if ((ignorePaymentPlan || ignoreCustomerId) && !customerHasCancelledGroupPlan) return;
+    if ((ignorePaymentPlan || ignoreCustomerId) && !customerHasCancelledGroupPlan) {
+      // member has been added to group plan but their subscription will not be changed
+      // automatically so they need a special message in the email
+      if (memberPlan.paymentMethod === this.constants.GOOGLE_PAYMENT_METHOD) {
+        previousSubscriptionType = 'Google';
```

Comment:
I am nervous about the constants being changed for any reason - for example, we might change UNLIMITED_CUSTOMER_ID from `habitrpg` to `habitica` at some point. That would then cause the Mandrill email templates to stop working correctly, unless we remembered to update them as well. Specifying the strings separately feels safer, however I don't feel very strongly about it so if you want me to use the constants, I will, and I'll add a comment to the code about updating the templates if the constants are ever changed.

The code would still be a bit ugly though. We'd have to do something like this to ensure that unexpected values were never passed through to the email templates:

```
      if (memberPlan.paymentMethod === this.constants.GOOGLE_PAYMENT_METHOD || memberPlan.paymentMethod === this.constants.IOS_PAYMENT_METHOD) {
        previousSubscriptionType = memberPlan.paymentMethod;
      } else if (memberPlan.customerId === this.constants.UNLIMITED_CUSTOMER_ID || memberPlan.customerId === this.constants.GROUP_PLAN_CUSTOMER_ID) {
        previousSubscriptionType = memberPlan.customerId;
      } else {
        // this triggers a generic message in the email template in case we forget
        // to update this code for new special cases
        previousSubscriptionType = 'unknown';
      }
```

Do you want me to go ahead with that?